cli/cmd/generate_aws_eks_audit.go

```diff
@@ -0,0 +1,917 @@
+				// we are going to write out new cached values, so if they run it - bail out - and run it again they'll get
+				// re-prompted.
+				if answer {
+					// Merge cached inputs to current options (current options win)
+					if err := mergo.Merge(GenerateAwsEksAuditCommandState, cachedOptions); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+					if err := mergo.Merge(GenerateAwsEksAuditCommandExtraState, extraState); err != nil {
+						return errors.Wrap(err, "failed to load saved options")
+					}
+				}
+			}
+
+			// Parse regions passed as part of the region cluster map
+			if len(GenerateAwsEksAuditCommandState.RegionClusterMap) > 0 {
+				// validate the format of supplied values is correct
+
+				awsParsedRegionClusterMap := make(map[string][]string)
+				for region, clusters := range GenerateAwsEksAuditCommandState.RegionClusterMap {
+					// verify each region is a valid aws region
+					if err := validateStringWithRegex(region, AwsEksAuditRegionRegex,
+						"invalid region name supplied"); err != nil {
+						return err
+					}
+					// parse the cluster comma seperated string into a list of clusters
+					parsedClusters := strings.Split(clusters, ",")
+					awsParsedRegionClusterMap[region] = append(awsParsedRegionClusterMap[region], parsedClusters...)
+				}
+				GenerateAwsEksAuditCommandState.ParsedRegionClusterMap = awsParsedRegionClusterMap
+			}
+
+			// Collect and/or confirm parameters
+			err = promptAwsEksAuditGenerate(GenerateAwsEksAuditCommandState, GenerateAwsEksAuditExistingRoleState,
+				GenerateAwsEksAuditCommandExtraState)
+			if err != nil {
+				return errors.Wrap(err, "collecting/confirming parameters")
+			}
+
+			return nil
+		},
+	}
+)
+
+type AwsEksAuditGenerateCommandExtraState struct {
+	AskAdvanced             bool
+	Output                  string
+	ConfigureBucketSettings bool
+	UseExistingKmsKey       bool
+	MultiRegion             bool
+	TerraformApply          bool
+}
+
+func (eks *AwsEksAuditGenerateCommandExtraState) isEmpty() bool {
+	return eks.Output == "" &&
+		!eks.AskAdvanced &&
+		!eks.ConfigureBucketSettings &&
+		!eks.UseExistingKmsKey &&
+		!eks.TerraformApply
+}
+
+// Flush current state of the struct to disk, provided it's not empty
+func (eks *AwsEksAuditGenerateCommandExtraState) writeCache() {
+	if !eks.isEmpty() {
+		cli.WriteAssetToCache(CachedAssetAwsEksAuditExtraState, time.Now().Add(time.Hour*1), eks)
+	}
+}
+
+func initGenerateAwsEksAuditTfCommandFlags() {
+	// add flags to sub commands
+	// TODO Share the help with the interactive generation
+	generateAwsEksAuditTfCommand.PersistentFlags().StringVar(
+		&GenerateAwsEksAuditCommandState.AwsProfile, "aws_profile", "", "specify aws profile")
+	generateAwsEksAuditTfCommand.PersistentFlags().BoolVar(
+		&GenerateAwsEksAuditCommandState.BucketEnableMfaDelete,
+		"enable_mfa_delete_s3",
+		false,
+		"enable mfa delete on s3 bucket. Requires bucket versioning.")
```

Comment:
I haven't noted capitalization of these acronyms in these audit messages. If the prevailing style in the codebase is to have these as lower case, it's fine. Otherwise, I think I'd make a pass at capitalizing throughout. i.e.: 

 ```suggestion
		"enable MFA delete on S3 bucket. Requires bucket versioning.")
```

Either way, just a nit.

Reply:
For consistency sake I'm going to leave as lower case